discovery-provider/src/tasks/index_user_bank_backfill.py

```diff
@@ -500,33 +502,54 @@ def process_user_bank_txs(stop_sig: str):
 
 
 def check_if_backfilling_complete(
-    session: Session, solana_client_manager: SolanaClientManager
+    session: Session, solana_client_manager: SolanaClientManager, redis: Redis
 ) -> bool:
-    stop_sig_tuple = (
-        session.query(IndexingCheckpoint.signature)
-        .filter(IndexingCheckpoint.tablename == index_user_bank_backfill_tablename)
-        .first()
-    )
-    if not stop_sig_tuple:
+    try:
+        redis_complete = redis.get(index_user_bank_backfill_complete)
+        if redis_complete:
+            redis_complete = str(redis_complete.decode())
+        if redis_complete == "true":
+            return True
+
+        stop_sig_tuple = (
+            session.query(IndexingCheckpoint.signature)
+            .filter(IndexingCheckpoint.tablename == index_user_bank_backfill_tablename)
+            .first()
+        )
+        if not stop_sig_tuple:
+            logger.error(
+                "index_user_bank_backfill.py | Tried to check if complete, but no stop_sig"
+            )
+            return False
+        else:
+            stop_sig = stop_sig_tuple[0]
+
+        one_sig_before_stop_result = solana_client_manager.get_signatures_for_address(
+            USER_BANK_ADDRESS,
+            before=stop_sig,
+            limit=1,
+        )
+        if one_sig_before_stop_result:
+            one_sig_before_stop_result = one_sig_before_stop_result["result"][0]
+            one_sig_before_stop = one_sig_before_stop_result["signature"]
+        else:
```

Comment:
nit: also maybe flip the if and the else, and in this case would only need the if statement which early returns, and the rest doesn't have to be nested

Reply:
brilliant